listings/applications/erc20/src/erc20_streaming.cairo

```diff
@@ -0,0 +1,128 @@
+#[starknet::contract]
+
+pub mod erc20_streaming {
+
+    // Import necessary modules and traits
+    use core::num::traits::Zero;
+    use starknet::get_caller_address;
+    use starknet::ContractAddress;
+    use starknet::LegacyMap;
+
+    #[storage]
+    struct Storage {
+        streams: LegacyMap<(ContractAddress, ContractAddress), Stream>,
+        erc20_token: ContractAddress,
```

Comment:
`erc20_token` can be moved to `Stream` struct, so each stream can use a specific erc20 contract